x/auth/ante.go

```diff
@@ -143,7 +143,7 @@ func validateAccNumAndSequence(ctx sdk.Context, accs []Account, sigs []StdSignat
 
 		accnum := accs[i].GetAccountNumber()
 		if ctx.BlockHeight() != 0 && accnum != sigs[i].AccountNumber {
```

Comment:
Wait how does this work, the signature shouldn't carry its own account number O_O

Reply:
huh?

Reply:
I don't understand the question either - the account number must be signed over.

Reply:
I agree it has to be signed over. However the StdSignature is whats being sent over the wire. You already have the address, so you can derive expected account number and sequence number from that. Its not even saving space, since to verify the signature you have to do this accnum check anyway.

Reply:
Verily. We should separate out what needs to be signed over and what is sent over the wire.

Reply:
I'll move this to a separate issue. I'm honestly a bit astounded that this is how it was implemented...

We definitely should spend more time ensuring all p2p data makes sense imo